src/number_theory/legendre_symbol/quadratic_reciprocity.lean

```diff
@@ -52,41 +52,30 @@ end
 
 /-- Euler's Criterion: a nonzero `a : zmod p` is a square if and only if `x ^ (p / 2) = 1`. -/
 lemma euler_criterion {a : zmod p} (ha : a ≠ 0) :
-  (∃ y : zmod p, y ^ 2 = a) ↔ a ^ (p / 2) = 1 :=
+  is_square (a : zmod p) ↔ a ^ (p / 2) = 1 :=
 begin
   apply (iff_congr _ (by simp [units.ext_iff])).mp (euler_criterion_units p (units.mk0 a ha)),
   simp only [units.ext_iff, sq, units.coe_mk0, units.coe_mul],
-  split, { rintro ⟨y, hy⟩, exact ⟨y, hy⟩ },
+  split, { rintro ⟨y, hy⟩, exact ⟨y, hy.symm⟩ },
   { rintro ⟨y, rfl⟩,
     have hy : y ≠ 0, { rintro rfl, simpa [zero_pow] using ha, },
     refine ⟨units.mk0 y hy, _⟩, simp, }
 end
 
 -- The following is used by number_theory/zsqrtd/gaussian_int.lean and archive/imo/imo2008_q3.lean
```

Comment:
This is somewhat off-topic, but this is a very weird doc comment. I've never seen a lemma talk about what other files it's being used in. Why is there a need to track this?

Reply:
I put this there as a reminder to myself that I have to check these two places when I make changes to this lemma. If this is considered bad style for mathlib, I can of course remove the comment (note that it's not part of the docstring).

Reply:
I personally consider this bad style, since it means that every time anyone else uses this theorem, three comment needs to be updated. Moreover, theorems shouldn't be changing regularly, and anyone refactoring a specific theorem should be Ctrl+F-ing the entirety of mathlib anyways.

Reply:
OK, I'm removing it. (It wasn't meant as a permanent fixture, just during the time I'm working on refactoring the Legendre symbol stuff.)